packages/shell-api/src/collection.ts

```diff
@@ -399,34 +399,29 @@ export default class Collection extends ShellApiClass {
   }
 
   @returnsPromise
-  async findAndModify(options: FindAndModifyMethodShellOptions = {}): Promise<Document> {
+  async findAndModify(options: FindAndModifyMethodShellOptions): Promise<Document> {
```

Comment:
Noticed that our types for `FindAndModifyMethodShellOptions['update']` allow `Document | Document[]` but unless I'm missing something I don't think that the refactor handles this case (and it doesn't seems like it should be supported anyway?). Maybe we should remove the `Document[]` part of the type in the same PR?

Reply:
No, you’re right – if it’s an array, we should pick the `Update` case rather than `Replace`. And it should be supported: https://docs.mongodb.com/manual/reference/method/db.collection.findAndModify/

> Starting in MongoDB 4.2, if passed an aggregation pipeline \[ <stage1>, <stage2>, ... \], db.collection.findAndModify() modifies the document per the pipeline.

Reply:
Ahhh, got it, thanks for the pointer! I got confused by [driver 4 doc](https://mongodb.github.io/node-mongodb-native/4.0/classes/collection.html#findoneandupdate) that doesn't specify `Document[]` as a possible option for `update`

Reply:
@gribnoysup Hm yeah … I guess the driver’s `Document` type just happily accepts arrays? https://www.typescriptlang.org/play?#code/JYOwLgpgTgZghgYwgAgCIHsHIN4ChkHIDaA1hAJ4BcyAzmFKAOYC61cI5A3LgL664J0IOsgCC1DFgC8xZpyA

Reply:
Haha, didn't checked before what `Document` type implementation looks like, but yeah, makes sense that array will match this very generous `Document` interface 😄